javalib/src/main/scala/java/nio/channels/spi/SelectorProvider.scala

```diff
@@ -0,0 +1,25 @@
+package java.nio.channels.spi
+
+import java.nio.channels._
+import java.net.ProtocolFamily
+
+abstract class SelectorProvider protected () {
+
+  def openDatagramChannel(): DatagramChannel
+  def openDatagramChannel(family: ProtocolFamily): DatagramChannel
+  def openPipe(): Pipe
+  def openSelector(): AbstractSelector
+  def openServerSocketChannel(): ServerSocketChannel
+  def openSocketChannel(): SocketChannel
+
+  // There is no JVM, so there is no way to inherit this, hence null
+  def inheritedChannel: Channel = null
+}
+
+object SelectorProvider {
+
+  // Scala Native does not support class loading, so we
```

Comment:
I think this comment is someplace between misleading and wrong.

I think the intent is that Scala Native support only the default
provider and not loading of alternate implementations.  Hard enough
to get one right. This is similar to how `FileSystemProvider` currently
only supports, depending upon the OS, `UnixFileSystem" or "WindowsFileSystem".

`Charset` and its unit-test gives an existence proof that one can 
dynamically load classes in Scala Native.
In some private sandbox work, which has not see action for six
months and is now paused, I added support for `ZipFileSystem`.

In my work of the past and future week, I am adding the loading
of various `RandomFactory` implementations.